apps/dashboard/src/components/engine/overview/import-backend-wallet-button.tsx

```diff
@@ -10,204 +9,327 @@ import {
+  const walletTypeOptions = supportsMultipleWalletTypes
+    ? EngineBackendWalletOptions
+    : [selectedOption];
+
+  const isAwsKmsConfigured = !!walletConfig.awsAccessKeyId;
+  const isGcpKmsConfigured = !!walletConfig.gcpKmsKeyRingId;
+
+  // Custom validation logic because required fields depend on the wallet type.
+  const values = form.getValues();
+  const isFormValid =
+    (walletType === "local" && values.privateKey) ||
+    (walletType === "aws-kms" && values.awsKmsArn) ||
+    (walletType === "gcp-kms" &&
+      values.gcpKmsKeyId &&
+      values.gcpKmsKeyVersionId);
 
   return (
     <>
       <Button onClick={() => setIsModalOpen(true)} variant="outline">
         Import
       </Button>
+
       <Dialog open={isModalOpen} onOpenChange={setIsModalOpen}>
         <DialogContent
           className="z-[10001] p-0"
           dialogOverlayClassName="z-[10000]"
         >
-          <form
-            onSubmit={form.handleSubmit((data) => {
-              const promise = importWallet.mutateAsync(data, {
-                onSuccess: () => {
-                  setIsModalOpen(false);
-                  trackEvent({
-                    category: "engine",
-                    action: "import-backend-wallet",
-                    label: "success",
-                    type: walletConfig?.type,
-                    instance: instanceUrl,
-                  });
-                },
-                onError: (error) => {
-                  trackEvent({
-                    category: "engine",
-                    action: "import-backend-wallet",
-                    label: "error",
-                    type: walletConfig?.type,
-                    instance: instanceUrl,
-                    error,
-                  });
-                },
-              });
+          <Form {...form}>
+            <form onSubmit={form.handleSubmit(onSubmit)}>
+              <div className="p-6">
+                <DialogHeader className="mb-4">
+                  <DialogTitle className="font-semibold text-2xl tracking-tight">
+                    Import wallet
+                  </DialogTitle>
+                </DialogHeader>
 
-              toast.promise(promise, {
-                success: "Wallet imported successfully",
-                error: "Failed to import wallet",
-              });
-            })}
-          >
-            <div className="p-6">
-              <DialogHeader className="mb-5">
-                <DialogTitle className="font-semibold text-2xl tracking-tight">
-                  Import {walletType} wallet
-                </DialogTitle>
-              </DialogHeader>
+                <div className="flex flex-col gap-5">
+                  {/* Wallet type */}
+                  <FormItem>
+                    <FormLabel>Wallet Type</FormLabel>
+                    <FormControl>
+                      <Select
+                        onValueChange={(value) =>
+                          setWalletType(value as EngineBackendWalletType)
```

Comment:
reset all the other keys here. 
It would be confusing to keep the data of previous wallet type ( like label )